src/class4Strings/StringAPI.java

```diff
@@ -0,0 +1,167 @@
+        return inputString;
+    }
+
+    //GET ANY INT VALUE
+    static int getInt() {
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Please enter any integer value");
+        while (!scanner.hasNextInt()) {
+            //Error message when entering non integer value
+            System.out.println("Please enter only integer value");
+            scanner.next();
+        }
+        int inputInt = scanner.nextInt();
+        //Error message when entering negative numeric value or zero
+        while (inputInt <= 0) {
+            System.out.println("Please enter only positive integer value");
+            //Error message when entering non integer value
+            while (!scanner.hasNextInt()) {
+                System.out.println("Please enter only integer value");
+                scanner.next();
+            }
+            inputInt = scanner.nextInt();
+        }
+        System.out.println("Int : " + inputInt);
+        return inputInt;
+    }
+
+    //FIND THE LETTER OF STRING WITH INPUT INDEX
+    static void findIndexLetter(String getString, int getInt) {
+        //Create array of char type
+        char[] ch = new char[getString.length()];
+        //Fill-in String elements in char array
+        for (int i = 0; i < getString.length(); ++i) {
+            ch[i] = getString.charAt(i);
+            if (i == getInt) {
+                System.out.println("The character at index " + i + " of your String is letter " + ch[i - 1]);
+                return;
+            }
+        }
+        System.out.println("Out of bounds");
+    }
+
+    //FIND NUMBER OF WORDS IN A TEXT
+    static void findNumberOfWords(String getString) {
+        int count = 0;
+        String[] startWord = getString.split(" ");
+        for (int i = 0; i < startWord.length; ++i) {
+            if (!startWord[i].isEmpty()) {
+                count++;
+            }
+        }
+        System.out.println("Number of words : " + (count));
+    }
+
+    //CHECK IF STRING IS PALINDROME
+    static void checkIfPalindrome(String getString) {
+        for (int i = 0, j = getString.length() - 1; i < getString.length(); ++i, --j) {
+            if (getString.charAt(i) != getString.charAt(j)) {
+                System.out.println("Not Palindrome");
+                return;
+            }
+        }
+        System.out.println("Palindrome");
+    }
+
+    //CHECK IF OPENING AND CLOSING PARANTHESIS ARE EQUAL
+    static void getParenthesis(String getString) {
+        for (int i = 0; i < getString.length(); ++i) {
+            //Error message when entering non parenthesis
+            if (getString.charAt(i) != '(' && getString.charAt(i) != ')') {
+                System.out.println("You are allowed to enter only parenthesis");
+                return;
+            }
+        }
+        int open = 0;
+        int close = 0;
+        //Count opening and closing parenthesis
+        for (int i = 0; i < getString.length(); ++i) {
```

Comment:
karair ham open-y ham close-y ham el invalid symbol-y mi hatik for-i mej aneir... else if-ov .. u ete hankarc for-ov fraluc handiper invalid symbol.. return aneir aseir senc chi lini))
du 2 for-ov es arel, chishty chishta, bayc avel gorcoxutyuna, hnenc el er kareli anel)